CONTRIBUTING.md

```diff
@@ -0,0 +1,900 @@
+
+```js
+// Good, code is self-descriptive
+var CARD_DECK_SIZE = 52
+
+// Shuffle the deck of cards.
+for (var i = 0; i < CARD_DECK_SIZE; i++) {
+	var j = i + (Math.random() * CARD_DECK_SIZE - i)|0
+
+	var tmp = deck[i]
+	deck[i] = deck[j]
+	deck[j] = tmp
+}
+
+// Also good, requires a descriptive comment.
+function toFastProperties(obj) {
+	// Bluebird's toFastProperties hack. Forces V8 to optimize object as prototype,
+	// significantly speeding up property access.
+
+	/* eslint-disable no-eval */
+	function C() {}
+	C.prototype = obj
+	new C()
+	return
+	eval(obj)
+	/* eslint-enable no-eval */
+}
+
+// Bad, redundant comments
+var CARD_DECK_SIZE = 52
+
+// Shuffle the deck of cards.
+function shuffle(deck) {
+	for (var i = 0; i < CARD_DECK_SIZE; i++) {
+		// Generate a random card index to swap at.
+		var j = i + (Math.random() * CARD_DECK_SIZE - i)|0
+
+		// Swap the cards at each index.
+		var tmp = deck[i]
+		deck[i] = deck[j]
+		deck[j] = tmp
+	}
+}
+```
+
+### Magic values
+
+Prefer constant variables to magic values where it helps with code readability. Also, don't use comments in place of a constant.
+
+```js
+// Good
+var CARD_DECK_SIZE = 52
+for (var i = 0; i < CARD_DECK_SIZE; i++) {
+	var j = i + (Math.random() * CARD_DECK_SIZE - i)|0
+	var tmp = a[i]
+	a[i] = a[j]
+	a[j] = tmp
+}
+
+// Bad, what's 52 for?
+for (var i = 0; i < 52; i++) {
+	var j = i + (Math.random() * 52 - i)|0
+	var tmp = a[i]
+	a[i] = a[j]
+	a[j] = tmp
+}
+
+// Also bad, comment not substitute for constant
+// 52 cards in a deck
+for (var i = 0; i < 52; i++) {
+	var j = i + (Math.random() * 52 - i)|0
+	var tmp = a[i]
+	a[i] = a[j]
+	a[j] = tmp
+}
+```
+
+### Distracting comments
+
+Don't comment *everything*. It's distracting, pointless, and a waste of time. Only comment when you need to.
+
+```js
+// Extremely bad. Don't *ever* do this.
+
+// This function loops through each item in a list, calling `f` with each item
+// and their respective index. If the function `f` returns explicitly with
+// `false`, iteration stops. This function returns the original list for
+// convenience.
+function forEach(list, f) {
+	// Loop through each entry in the list with the loop index `i`
+	for (var i = 0; i < list.length; i++) {
+		// Call the function with the current item and index
+		if (f(list[i], i) === false) {
+			// If the function explicitly returns `false`, immediately stop
+			// iteration.
+			break
+		}
+	}
+	// Return the list, for convenience.
+	return list
+}
+```
+
+### Initial whitespace
+
+Start your comments with a space. It's more readable.
+
+```js
+// Good
+// This is a comment.
+
+// Bad
+//This is a comment.
+```
+
+### Grammar and spelling
+
+Use proper grammar and spelling in your comments. This shouldn't need explanation.
```

Comment:
Barrier of entry to non-english speakers. Better to let it slide and fix after the fact.